File: java/java-arithmetic/src/main/java/com/leetcode/first/J66.java
package com.leetcode.first;

import java.util.Arrays;

public class J66 {
    public static void main(String[] args) {
        System.out.println(Arrays.stream(new J66().plusOne(new int[]{9})));
    }

    public int[] plusOne(int[] digits) {
        int right=digits.length -1;

        while (right>=0){
            int num =digits[right];
            if (num+1>=10){
                digits[right]= (num+1)% 10;
            }else{
                digits[right]=num+1;
                return digits;
            }
            right--;
        }

        digits=new int[digits.length+1];
        digits[0] =1;
        return digits;
    }

}

File: test-python/leetcode/J66.py
class Solution:
    def plusOne(self, digits: list[int]) -> list[int]:
        n = len(digits)-1
        for i in range(n,-1,-1):
            num = digits[i]
            a = num +1
            if a < 10 :
                digits[i] = a
                return digits
            else:
                digits[i]=0

        result=[]
        for i in range(len(digits)+1):
            if i == 0:
                result.append(1)
            else:
                result.append(0)
        
        return result




Solution().plusOne(digits=[1,2,3])